### PR TITLE
docs: Add issue template

### LINKED ISSUE #2805: Use issue template forms for issues
# What could be improved
Currently you have to manually edit a markdown template text when submitting issues, we should configure so that we are using the issue form feature instead.

# Why should this be improved
Easier for our users to file issues and less chance that they miss something, or just remove the whole template.

# Any risks?
That it is less versatile then our current solution.

# More information
https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms

### PR DESCRIPTION
**Pull request by clairecharles**
# Description

This fixes #2805
I added a template to the issue form so you won't have to fill in the headers for an issue.

## Checklist

- [X] I have followed the [Contributor Guide] when preparing my PR.
- [-] I have updated/added tests for ALL new/updated/fixed functionality.
- [X] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [-] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [ ] No, this PR is not a breaking change.

## Related Issues

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

**Comment by spydon** [comment]
Initially looks good, it should be updated to reflect what is in here though:
https://raw.githubusercontent.com/flame-engine/flame/main/.github/ISSUE_TEMPLATE/1_bug.md

And octocat should not be the default assignee. :joy:

File: .github/ISSUE_TEMPLATE/config.yml [hunk 1]
```diff
@@ -1,8 +1,66 @@
...
+  - type: markdown
+    attributes:
+      value: |
+        When reporting a bug, please read this complete template and fill in all the questions in order to get a better response!
+ - type: textarea
+    id: what-happened
+    attributes:
+      label: What happened?
+      description: Tell us, what happened?
+      placeholder: Tell us what you see!
```
Comment (spydon) [inline .github/ISSUE_TEMPLATE/config.yml:16]: I don't think there is need for a placeholder here

File: .github/ISSUE_TEMPLATE/config.yml [hunk 2]
```diff
@@ -1,8 +1,66 @@
...
+    validations:
+      required: true
+      
+  - type: textarea
+    id: expectation
+    attributes:
+      label: What do you expect?
+      description: Also tell us, what behavior did you expect? 
+      placeholder: Tell us what you want!
```
Comment (spydon) [inline .github/ISSUE_TEMPLATE/config.yml:25]: I don't think there is need for a placeholder here

File: .github/ISSUE_TEMPLATE/config.yml [hunk 3]
```diff
@@ -1,8 +1,66 @@
...
+    validations:
+      required: false
```
Comment (spydon) [inline .github/ISSUE_TEMPLATE/config.yml:27]: Should also be required

File: .github/ISSUE_TEMPLATE/config.yml [hunk 4]
```diff
@@ -1,8 +1,66 @@
...
+      description: If you have any debug / error logging, please fill it here within the code block below
+      render: shell
+  - type: checkboxes
+    id: terms
+    attributes:
+      label: Code of Conduct
+      description: By submitting this issue, you agree to follow our [Code of Conduct](https://example.com)
+      options:
+        - label: I agree to follow this project's Code of Conduct
+          required: true
```
Comment (spydon) [inline .github/ISSUE_TEMPLATE/config.yml:42]: This section can be removed

File: .github/ISSUE_TEMPLATE/config.yml [hunk 5]
```diff
@@ -1,8 +1,66 @@
...
+Create a list of more environment information, like:
+* Flame version: 1.0.0
+* Platform affected: android, ios, web
+* Platform version affected: android 9, ios 13
+-->
+
+# More information
+<!-- Do you have any other useful information about this bug report? Please write it down here -->
+<!-- Possible helpful information: references to other sites/repositories -->
+<!-- Are you interested in working on a PR for this? -->
```
Comment (spydon) [inline .github/ISSUE_TEMPLATE/config.yml:66]: These questions should be part of the form

**Comment by clairecharles** [comment]
Hi @spydon I have made changes to the file. Could you please look into it :)

File: .github/ISSUE_TEMPLATE/config.yml [hunk 6]
```diff
@@ -1,8 +1,93 @@
...
+      label: Relevant log output
+      description: If you have any debug / error logging, please fill it here within the code block below
+      render: shell
+
+
+# Flutter doctor output
+Execute in a terminal and put output into the code block below 
+```
+Output of: flutter doctor -v
+```
```
[the remark this answers is not in the document]
Comment (spydon) [reply .github/ISSUE_TEMPLATE/config.yml:65]: And similarly on all other markdown that is left underneath, so everything that starts with a `#` should have its own section that starts with `type:`

**Comment by clairecharles** [comment]
Made a few changes :)

File: .github/ISSUE_TEMPLATE/config.yml [hunk 7]
```diff
@@ -1,8 +1,96 @@
-blank_issues_enabled: true
-contact_links:
-  - name: I have some questions about Flame.
-    url: https://stackoverflow.com/tags/flame
-    about: Ask your questions on StackOverflow! The community is always willing to help out.
-  - name: Join us on Discord!
-    url: https://discord.gg/pxrBmy4
-    about: Ask your questions in our community!
+name: Bug Report
```
Comment (anleac) [inline .github/ISSUE_TEMPLATE/config.yml:1]: NIT: I'm not sure if this is the changeset we want; the change of the markdown looks solid 🙇 But I don't believ we should be modifying the `config.yml` but instead be creating a new file, `bug_report.yml` for example, in this same folder with the contents you've added (each new template/form should be it's own unique file).

More info here: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms

Comment (clairecharles) [reply .github/ISSUE_TEMPLATE/config.yml]: Hi @anleac does it look good now?

**Review by spydon — APPROVED** [review]
Good job @clairecharles!

**Pull request merged**